includes/REST_API/Link_Controller.php

```diff
@@ -136,8 +132,8 @@ public function parse_link( $request ) {
 		$response = wp_safe_remote_get( $url, $args );
 
 		if ( WP_Http::OK !== wp_remote_retrieve_response_code( $response ) ) {
-			// Not saving the error response to cache since the error might be temporary.
-			return new WP_Error( 'rest_invalid_url', get_status_header_desc( 404 ), [ 'status' => 404 ] );
```

Comment:
I think this error should remain. If the site returns a 404 or 500 error, that is a error and the user should know about this. Linking to the broken response is a bad idea.

Reply:
Could still be a temporary issue.

As per discussion on #1966, we don't want to show errors in that cases. I recommend chiming in there if you feel strongly otherwise.